### ISSUE TITLE
feature request: make settings UI highlight chosen settings when navigating

### ISSUE THREAD

**Issue by briaguya0** [Mr-Wiseguy/Zelda64Recomp#367]
this is less of a specific bug and more of a suggestion for how to handle the somewhat awkward ux when moving between settings with keyboard/controller. it is not clear to me how the next highlighted option is determined when moving between a slider option and a radio option

one example is at the very top of the general settings window, here is a video of me pressing "down, down, up, up" repeatedly

[Screencast from 2024-06-05 01-32-19.webm](https://github.com/Zelda64Recomp/Zelda64Recomp/assets/70942617/b29844a8-bb93-4ce7-a599-cfb69887c154)

and here is an example of navigating between "joystick deadzone" and "background input" (i'm pressing "down, right, up" repeatedly)

[Screencast from 2024-06-05 01-35-41.webm](https://github.com/Zelda64Recomp/Zelda64Recomp/assets/70942617/4844626a-3691-42cf-a88d-6b708d455d6b)

i'd like to suggest that instead of trying to keep track of where the user navigates from horizontally, navigating between options puts the highlight on the chosen option, so for

![image](https://github.com/Zelda64Recomp/Zelda64Recomp/assets/70942617/1793f92f-109c-404c-aedf-836f9468fab0)

instead of pressing down repeatedly leading to highlighting
```
Switch
Rumble Slider
Gyro Slider
Mouse Slider
Deadzone Slider
On
On
None
On
None
```
it would lead to highlighting
```
Switch
Rumble Slider
Gyro Slider
Mouse Slider
Deadzone Slider
On
Off
Invert Y
Off
None
```

**Comment by Mr-Wiseguy** [comment]
I agree with this change, and in fact it's already going to be added as a side-effect of a UI refactor that's being worked on for modding support. The refactor will turn the various types of menu options into reusable components so the menu can be code-driven (which will let mods add their own menu options). The menu element generation code will also handle setting up navigation directions automatically, which we've decided should work this way.

**Comment by Reonu** [comment]
This is now in-place for mod config menus. It'll be added to the menus of the base project in the future.
